microcloud/service/lxd.go

```diff
@@ -680,6 +682,82 @@ func (s *LXDService) SetConfig(target string, secret string, config map[string]s
 	return c.UpdateServer(newServer, "")
 }
 
+// Restart requests LXD to shutdown, then waits until it is ready.
+func (s *LXDService) Restart(timeoutSeconds int) error {
+	c, err := s.client("")
+	if err != nil {
+		return err
+	}
+
+	_, _, err = c.RawQuery("PUT", "/internal/shutdown", nil, "")
```

Comment:
Does this actually cause the snap part of lxd to restart? Can u point me to which bit is triggered in snap packaging?

Reply:
Yeah, the `daemon.start` hook does actually run again when this happens.

If I'm understanding this correctly, the snap sets up a systemd service based on the the config [here](https://github.com/canonical/lxd-pkg-snap/blob/latest-edge/snapcraft.yaml#L71-L77). 

Seems that a restart is triggered `on-failure`. I assume this is triggered because LXD is getting restarted directly and not through systemd.

Reply:
lxd stops itself and systemd starts it again

Reply:
Ah so it's also enabled by default